include/ofxColour.h

```diff
@@ -0,0 +1,175 @@
+
+/* String used to indicate that no colour management is available. */
+#define kOfxImageEffectPropColourManagementNone "OfxImageEffectPropColourManagementNone"
+/* String used to indicate that Native colour management is available. */
+#define kOfxImageEffectPropColourManagementNative "OfxImageEffectPropColourManagementNative"
+/* String used to indicate that OCIO colour management is available. */
+#define kOfxImageEffectPropColourManagementOCIO "OfxImageEffectPropColourManagementOCIO"
+
+/** @brief The path to the OCIO config used for this instance
+
+   - Type - string X 1
+   - Property Set - image effect instance (read only)
+   - Valid Values - Filesystem path to the config or URI starting ocio://
+
+A host must set this property on any effect instances where it has negotiated 
+OCIO colour management (kOfxImageEffectPropColourManagementOCIO).
+Use of URIs for built-in configs, such as ocio://default is permitted.
+*/
+#define kOfxImageEffectPropOCIOConfig "OfxImageEffectPropOCIOConfig"
+
+/** @brief The colourspace used for this clip
+
+   - Type - string X 1
+   - Property Set - clip instance (read/write)
+   - Valid Values - colourspace that is permitted under the style in use
+
+Hosts should set this property to the colourspace of the input clip. Typically 
+it will be set to the working colourspace of the host but could be any valid 
+colourspace.
+Plug-ins may set this property on an output clip. Plug-ins which output motion 
+vectors or similar images which should not be colour managed can use the data 
+colourspace which is present in the built-in OCIO configs.
+Both host and plug-in should use the value of 
+kOfxImageClipPropPreferredColourspace where reasonable.
+*/
+#define kOfxImageClipPropColourspace "OfxImageClipPropColourspace"
+
+/** @brief The preferred colourspace for this clip
+
+   - Type - string X N
+   - Property Set - clip instance (read only) and ::kOfxImageEffectActionGetClipPreferences action out args property (read/write)
+   - Valid Values - colourspace that is permitted under the style in use. 
+                    For Native, this any scene colourspace or role from ofxColourspaceList.h.
+                    For OCIO, this could be any string acceptable to Config::getColorSpace().
+
+Plug-ins may set this property during kOfxImageEffectActionGetClipPreferences 
+to request images in a colourspace which is convenient for them. The
+property is an ordered set of colourspace identifiers, which may be any of the
+names or aliases supported by the colour management style in use. If plug-ins
+prefer more esoteric colourspaces, they are encouraged to also include generic
+roles as a fallback. For example a colour grading plug-in which supports a
+specific camera and expects a log colourspace might list:
+
+"arri_logc4", "arri_logc3_ei800", "ACEScct", "color_timing"
+
+The host should provide input images in the first mutually agreeable
```

Comment:
"the first mutually agreeable" is I hope deliberately vague. A host might decide it is more agreeable to offer a display-referred colour space from later in the plugin's list of spaces than to convert from a display-referred space back to a scene-referred space earlier in the plugin's list.

Reply:
That is exactly what I had in mind.